app/containers/Avatar/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Q } from '@nozbe/watermelondb';
+
+import database from '../../lib/database';
+import { getUserSelector } from '../../selectors/login';
+import Component from './Avatar';
+
+const Avatar = React.memo(({ type, text, ...props }) => {
+	const [avatarETag, setAvatarETag] = useState();
+
+	useEffect(() => {
+		let subscription;
+		(async() => {
```

Comment:
Open to suggestions about how we can handle the unsubscribe without IIFE

Reply:
There's an issue here that `subscription` won't have a `unsubscribe` method even using IIFE.
Correct me if I'm wrong, but as this unnamed function doesn't `await`, return is called right away.

This could be rewritten as a normal function and being called right after, but the problem would remain.
```js
// taken from https://github.com/facebook/react/issues/14326#issuecomment-441680293
useEffect(() => {
  async function fetchMyAPI() {
    let url = 'http://something';
    let config = {};
    const response = await myFetch(url);
    console.log(response);
  }  

  fetchMyAPI();
}, []);
```

The issue here is that we need `subscription` to return a value on `useEffect`, so the unmount cleans everything.
If you switch to React.Component, this issue is gone easily :)